alphagov/paas-cf
# [#116974545] Split apps and system services entry endpoints traffic

[#116974545 Seperate IP Restrictions for Apps and System domains](https://www.pivotaltracker.com/story/show/116974545)

## What

As an operator I want to be able to restrict the IPs that can connect to the CF API endpoints without affecting public access to tenant deployed applications.

During kick-off we decided to solve this by using ELBs for each one of the exposed CF API services which are being handled by the routers.

## How to review

This story should be review in this context:

 * Create a new security group for CF related APIs, which is open to the office only.
 * Create dedicated ELBs for: CloudController, UAA, Loggregator and Doppler.
 * Configure these ELBs in BOSH so the instances get automatically registered on creation.
 * Configure the DNS for those services, dropping the wildcard DNS and adding specific ones explicitelly. 
 * Additionally (following [boy scout rule](http://programmer.97things.oreilly.com/wiki/index.php/The_Boy_Scout_Rule)): I want to clean up, reorganise and rename the terraform config and resources related to ELBs and Security groups.

## How to test

 1. Deploy the environment and run the smoke tests. Smoke tests is enough, as it uses all the affected endpoints.
 2. Try to deploy an app manually from the office and operate it with `cf-cli`

## Notes

 * This change is disruptive and will stop the platform for working as being deployed.
 * You might end in a situation with negative DNS caching in concourse. Wait 5m if so.
 * The domains `*.uaa.${system_root_domain}` and `*.login.${system_root_domain}` did not seem to be used.

## Who can review

anyone but @keymon or @alext
